vignettes/Age_metanalysis_vignette.Rmd

```diff
@@ -0,0 +1,380 @@
+    
+    #FDR-correction with Benjamini-Hochberg method for Wald p-values
+    final_df <- as.matrix(cbind(R_List,
+                                wald_list,
+                                p.adjust(wald_list,method = "BH")))
+    
+    #Finalize results for the single dataset
+    colnames(final_df) <- c(paste0(dataset,"_Correlation"),
+                            paste0(dataset,"_pvalue"),
+                            paste0(dataset,"_Qvalue"))
+    #final_df <-na.omit(final_df)
+    rownames(final_df) <- species
+    return(final_df)
+  
+    }
+
+
+    linear_models <- lapply(tse_datasets, singleDatasetAnalyze)
+    names(linear_models) <- tse_datasets
+    
+    return(linear_models)
+
+} 
+
+```
+
+
+
+
+
+Now that we have computed effect-sizes for each population, we can meta-analyze each microbial feature. In order to do this, we define a function applying the “metacor” function from the package “metafor”.
+```{r}
+runMetaanalysis <- function(R_vector, n) {
+  a <- meta::metacor(cor=R_vector,
+                     n=n,
+                     studlab=rownames(R_vector),
+                     method.tau="PM",
+                     sm="ZCOR")
+  
+  final_vector <-c(a$TE.random,
+                   a$seTE.random,
+                   paste(a$lower.random,a$upper.random,sep=";"),
+                   a$zval.random,
+                   a$pval.random,
+                   a$tau2,
+                   a$I2)
+  
+  names(final_vector) <- c("RE_Correlation","SE_RE","CI_RE","Zscore","p-value","tau2","I^2")
+  return(final_vector)
+}
+```
+
+
+
+## Application of the described procedure in the meta-analysis of age-related microbial species
+
+```{r}
+#Computing Correlation (R) of species relative abundances with age for all datasets 
+Corr_list <- computeStandardizedMeanDifference(species_tse)  
+
+
+#Merging outputs of all datasets so to have a single dataframe with all the 
+#species found across the cohorts
+final_df <- Reduce(function(x, y) merge(x, y, all=TRUE), 
+                   lapply(Corr_list, function(x) data.frame(x, rn = row.names(x))))
+
+final_df <- final_df  %>% column_to_rownames(var="rn")
+
+
+#Dataframes with random effect and respective vector containing sample size of 
+#each dataset
+d_matrix <- final_df %>% 
+  select(contains("Correlation"))
+d_matrix <- t(d_matrix)
+
+
+n_vector <- as.data.frame(colData(species_tse)) %>% 
+  group_by(study_name) %>% 
+  summarize(N= n())
+
+
+n_vector <- n_vector[match(str_replace(rownames(d_matrix), "_Correlation",""),
+                           n_vector$study_name ),]$N
+
+meta_analysis <- t(as.data.frame(apply(FUN=function(x) {runMetaanalysis(x,n_vector)},
+                                       MARGIN=2,
+                                       d_matrix)))
+  
+final_df <- cbind(final_df, meta_analysis)
+final_df$FDR_Qvalue <- p.adjust(final_df$`p-value`,method = "BH")
+```
+
+```{r, echo=FALSE, message=FALSE}
+filter_sort <- function(df_to_sort, na_filter){
```

Comment:
This function gets a little complicated and I'm not sure what it does, but if it works then it's good! If it has any general use then it may be also worth documenting in the R/ directory.